hub/core/storage/lru_cache.py

```diff
@@ -408,7 +408,7 @@ def _all_keys(self):
         key_set = set()
         if self.next_storage is not None:
             key_set = self.next_storage._all_keys()  # type: ignore
-        key_set = key_set.union(self.cache_storage._all_keys())
+        key_set = set().union(key_set, self.cache_storage._all_keys())
```

Comment:
```suggestion
        key_set = key_set.union(self.cache_storage._all_keys())
```
the original seems better, here we're unnecessarily adding an empty set to the union